### PR TITLE
Dartanalyzer, part 2

### PR DESCRIPTION
**Pull request by hansl**
@jelbourn  Please review as your convenience.

This implements all the code for dartanalyzer to run successfully. There are too many errors in Dart files right now for me to run the analyzer on Travis (it would break, a lot). I will make a separate PR to fix errors and enable it on travis.

Also, please note there was a bug in Broccoli-Dest-Copy which this fixes.

### CODE REVIEW

File: ember-cli-build.js [hunk 1]
```diff
@@ -18,11 +22,11 @@ module.exports = function(defaults) {
   var demoAppCssTree = new BroccoliSass(['src/demo-app'], './demo-app.scss', 'demo-app/demo-app.css');
   var demoCssTree = getCssTree('demo-app');
   var componentCssTree = getCssTree('components');
-  var angularAppTree = new Angular2App(defaults);
+  var angularAppTree = new Angular2App(defaults).toTree();
```
Comment (jelbourn) [inline ember-cli-build.js]: What is the `toTree()` for?

**Comment by jelbourn** [comment]
Can you make the dart travis build optional temporarily so that it isn't red for PRs?

File: ember-cli-build.js (hunk 1 above)
Comment (hansl) [inline ember-cli-build.js]: It returns a Broccoli tree. It was called below before, but in an iteration I needed access to it. I'll undo that.

**Comment by hansl** [comment]
The travis build is not runnin g the dartanalyzer yet, so it should pass. I think I fixed it.

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**